### PR TITLE
[SPIR-V] Add support for the SPIR-V extension SPV_INTEL_bfloat16_conversion

### PR DESCRIPTION
**Pull request by VyacheslavLevytskyy**
This PR is to add support for the SPIR-V extension SPV_INTEL_bfloat16_conversion (https://github.com/KhronosGroup/SPIRV-Registry/blob/main/extensions/INTEL/SPV_INTEL_bfloat16_conversion.asciidoc) and OpenCL extension cl_intel_bfloat16_conversions (https://registry.khronos.org/OpenCL/extensions/intel/cl_intel_bfloat16_conversions.html).

### CODE REVIEW

File: llvm/test/CodeGen/SPIRV/extensions/SPV_INTEL_bfloat16_conversion/bfloat16-conv-negative2.ll [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+target datalayout = "e-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024-n8:16:32:64"
+target triple = "spir64-unknown-unknown"
+
+define spir_func void @test(<8 x float> %in) {
+  %res = tail call spir_func <4 x i16> @_Z27__spirv_ConvertFToBF16INTELDv8_f(<8 x float> %in)
+  ret void
+}
+
+declare spir_func <4 x i16> @_Z27__spirv_ConvertFToBF16INTELDv8_f(<8 x float>)
```
Comment (michalpaszkowski) [inline llvm/test/CodeGen/SPIRV/extensions/SPV_INTEL_bfloat16_conversion/bfloat16-conv-negative2.ll:12]: Please add a missing line at the end of the file

Comment (VyacheslavLevytskyy) [reply llvm/test/CodeGen/SPIRV/extensions/SPV_INTEL_bfloat16_conversion/bfloat16-conv-negative2.ll:12]: Thank you

**Review by michalpaszkowski — APPROVED** [review]
Thank you! LGTM!

**Comment by MaskRay** [comment]
This commit is associated with a `users.noreply.github.com` email address. Per 
https://discourse.llvm.org/t/hidden-emails-on-github-should-we-do-something-about-it/74223/32 , proper email addresses are preferred.

**Pull request merged**